packages/app/src/components/Synchronization/SynchronizationAPIProvider.tsx

```diff
@@ -0,0 +1,48 @@
+/*---------------------------------------------------------------------------------------------
+ * Copyright (c) Bentley Systems, Incorporated. All rights reserved.
+ * See LICENSE.md in the project root for license terms and full copyright notice.
+ *
+ * This code is for demonstration purposes and should not be considered production ready.
+ *--------------------------------------------------------------------------------------------*/
+import { Text } from "@itwin/itwinui-react";
+import { navigate, RouteComponentProps } from "@reach/router";
+import React, { PropsWithChildren } from "react";
+
+import { SynchronizationClient } from "../../api/synchronization/synchronizationClient";
+import { useApiPrefix } from "../../api/useApiPrefix";
+
+type SynchronizationAPIProviderProps = PropsWithChildren<{
+  accessToken: string;
+}>;
+
+export const SynchronizationAPIProvider = ({
+  accessToken,
+  children,
+}: RouteComponentProps & SynchronizationAPIProviderProps) => {
+  const urlPrefix = useApiPrefix();
+  const [isAuthorized, setIsAuthorized] = React.useState(false);
+
+  React.useEffect(() => {
+    if (!accessToken) {
+      return;
+    }
+
+    const client = new SynchronizationClient(urlPrefix, accessToken);
+    client
+      .getAuthorization(window.location.href)
+      .then(({ authorizationInformation }) => {
+        if (
+          !authorizationInformation?.isUserAuthorized &&
+          authorizationInformation?._links?.authorizationUrl?.href
+        ) {
+          return navigate(
```

Comment:
dont need to set isAuthorized before we navigate to this url(which i assume is the demo portal)?
and i guess after the "rerdirect" itll then set authorized?

Reply:
nvm, discussed this with raph